unittest/gunit/components/percona_telemetry/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2024 Percona LLC and/or its affiliates. All rights reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2 of
+# the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+   
+MY_CHECK_CXX_COMPILER_WARNING("-Wno-unused-local-typedefs" HAS_WARN_FLAG)
+IF(HAS_WARN_FLAG)
+  STRING_APPEND(CMAKE_CXX_FLAGS " ${HAS_WARN_FLAG}")
+ENDIF()
+
+INCLUDE_DIRECTORIES(SYSTEM
```

Comment:
Better use `target_include_directories` for individual targets

Reply:
I followed the way other unit tests do it. The idea is that each directory of UT can have several target binaries (a loop ath the end of CMakeLists.txt). All of these targets should have common include paths. So they are specified as INCLUDE_DIRECTORIES (for all) instead of TARGET_INCLUDE_DIRECTORIES for every particular target.

We could add `TARGET_INCLUDE_DIRECTORIES(${test}-t PUBLIC ${GMOCK_INCLUDE_DIRS})` in the loop at the end, but it has the same effect as INCLUDE_DIRECTORIES

Reply:
My comment mostly comes from the Modern CMake best practices - always use target-based directives. I believe the main idea behind them is to avoid CMake context pollution. For instance, if you do `INCLUDE_DIRECTORIES` here, then every other `CMakeLists.txt` file you directly or indirectly (via subdirs, for instance) include from this one will inherit these new includes (even when they are not necessarily needed). But I agree, in this particular case, we are not doing anything like this and this does not really matter - you can ignore my comment.